### PR TITLE
Fix: Unpublish parameter updated to use state_selected

### PR DESCRIPTION
**Pull request by Bernardvdv**

### CODE REVIEW

**Review by adam-murray — CHANGES_REQUESTED** [review]
Would it be possible to add a test which sends a request to the unpublished endpoint, to verify the error is no longer thrown?

File: tests/test_integrations.py [hunk 1]
```diff
@@ -123,3 +127,19 @@ def test_nested_app_references(self):
...
+class VersioningIntegrationTestCase(CMSTestCase):
+    def test_unpublish_versioned_item(self):
+        """
+        When unpublishing a versioned item, the state should be changed to unpublish
+        """
+        user = self.get_superuser()
+        version = PageVersionFactory(
+            content__title="test", content__language="en", state=PUBLISHED
+        )
+        # Changing version state to unpublished
```
Comment (adam-murray) [inline tests/test_integrations.py:141]:
```suggestion

        # Changing version state to unpublished
```

Comment (adam-murray) [reply tests/test_integrations.py:142]: Just one minor change to keep it consistent with the AAA formatting.

**Review by adam-murray — APPROVED** [review]

File: tests/test_integrations.py [hunk 2]
```diff
@@ -123,3 +127,20 @@ def test_nested_app_references(self):
         self.assertContains(response, "pagecontent")
         self.assertContains(response, get_object_preview_url(page_content))
         self.assertContains(response, page_content.versions.first().state)
+
+
+class VersioningIntegrationTestCase(CMSTestCase):
+    def test_unpublish_versioned_item(self):
```
Comment (Aiky30) [inline tests/test_integrations.py:133]: I'm confused by what this test is actually checking for. The filter covers that when versions are set to certain states they are hidden / shown. What is the purpose / aim of this test?

I would have thought that a test for the state selected would be calling get_all_reference_objects

Comment (Bernardvdv) [reply tests/test_integrations.py]: @Aiky30 Removed the test as we already have a test that confirms unpublishing still works as expected 

https://github.com/Bernardvdv/djangocms-references/blob/bug/unpublish_parameter_updated/tests/test_cms_config.py#L142

**Review by Aiky30 — APPROVED** [review]

**Pull request merged**